### PR TITLE
feat, ruby: enable oauth client generation

### PR DESCRIPTION
**Pull request by armandobelardo**
Modeled after https://github.com/fern-api/fern/pull/3764

This also removes literals from method signatures and hard codes them into their corresponding fields within the request construction

### CODE REVIEW

File: generators/ruby/sdk/src/ClientsGenerator.ts [hunk 1]
```diff
@@ -374,6 +421,41 @@ export class ClientsGenerator {
...
+            const accessTokenClass = new AccessToken(this.clientName, this.oauthScheme.configuration);
+
+            let maybeRefereshTokenDetails;
+            if (this.oauthScheme.configuration.refreshEndpoint) {
+                maybeRefereshTokenDetails = getOauthRefreshTokenFunctionMetadata({
+                    refreshEndpoint: this.oauthScheme.configuration.refreshEndpoint,
+                    artifactRegistry: this.artifactRegistry
+                });
+            }
+            const fullRubyDetails = {
```
Comment (amckinney) [inline generators/ruby/sdk/src/ClientsGenerator.ts:435]: nit: maybe best to just name this `oauthConfiguration` so its consistent with the property name used below?

File: seed/ruby-sdk/oauth-client-credentials-environment-variables/lib/fern_oauth_client_credentials_environment_variables.rb [hunk 2]
```diff
@@ -12,14 +13,24 @@ class Client
...
     # @param timeout_in_seconds [Long]
-    # @param token [String]
+    # @param client_id [String]
+    # @param client_secret [String]
     # @return [SeedOauthClientCredentialsEnvironmentVariablesClient::Client]
-    def initialize(token:, base_url: nil, max_retries: nil, timeout_in_seconds: nil)
+    def initialize(base_url: nil, max_retries: nil, timeout_in_seconds: nil, client_id: nil, client_secret: nil)
+      @oauth_provider = SeedOauthClientCredentialsEnvironmentVariablesClient::OauthTokenProvider.new(
+        client_id: client_id,
+        client_secret: client_secret,
```
Comment (amckinney) [inline seed/ruby-sdk/oauth-client-credentials-environment-variables/lib/fern_oauth_client_credentials_environment_variables.rb:22]: Does this support reading the variables from the environment yet? Totally fine if we leave that for a follow-up if not.

**Review by amckinney — APPROVED** [review]
Mainly focused on the generated snapshots - functionality looks good to me.

File: seed/ruby-sdk/oauth-client-credentials-environment-variables/lib/fern_oauth_client_credentials_environment_variables.rb (hunk 2 above)
Comment (armandobelardo) [reply seed/ruby-sdk/oauth-client-credentials-environment-variables/lib/fern_oauth_client_credentials_environment_variables.rb:22]: ah yes forgot to add, will do that now

Comment (armandobelardo) [reply seed/ruby-sdk/oauth-client-credentials-environment-variables/lib/fern_oauth_client_credentials_environment_variables.rb:23]: added!

**Pull request merged**